crates/fuel-core/src/schema/tx.rs

```diff
@@ -276,16 +277,48 @@ impl TxMutation {
         gas_price: Option<U64>,
     ) -> async_graphql::Result<Vec<DryRunTransactionExecutionStatus>> {
         let block_producer = ctx.data_unchecked::<BlockProducer>();
-        let params = ctx
+        let consensus_params = ctx
             .data_unchecked::<ConsensusProvider>()
             .latest_consensus_params();
+        let block_gas_limit = consensus_params.block_gas_limit();
 
+        let mut max_gas_usable: u64 = 0;
         let mut transactions = txs
             .iter()
             .map(|tx| FuelTx::from_bytes(&tx.0))
             .collect::<Result<Vec<FuelTx>, _>>()?;
+        let fee = consensus_params.fee_params();
         for transaction in &mut transactions {
-            transaction.precompute(&params.chain_id())?;
+            match transaction {
+                FuelTx::Script(tx) => {
+                    max_gas_usable = max_gas_usable
+                        .saturating_add(tx.max_gas(consensus_params.gas_costs(), fee));
+                }
+                FuelTx::Create(tx) => {
+                    max_gas_usable = max_gas_usable
+                        .saturating_add(tx.max_gas(consensus_params.gas_costs(), fee));
+                }
+                FuelTx::Mint(_) => {
+                    // Mint transactions doesn't consume gas and so we will add a flat 10% of the block gas limit
+                    max_gas_usable = max_gas_usable.saturating_add(block_gas_limit / 10);
```

Comment:
People shouldn't submit `Mint` transaction. It is only generated by a block producer. So we need to return an error=)